io-sim-classes/src/Control/Monad/Class/MonadThrow.hs

```diff
@@ -147,6 +131,13 @@ class MonadThrow m => MonadCatch m where
 --
 data HandlerM m a = forall e. Exception e => HandlerM (e -> m a) 
 
+deriving instance (Functor m) => Functor (HandlerM m)
+
+-- | Like 'catches' but for 'MonadCatch' rather than only 'IO'.
+--
+catchesM :: forall m a. MonadCatch m
+         => m a -> [HandlerM m a] -> m a
+catchesM ma handlers = ma `catch` catchesHandler handlers
```

Comment:
Yes, this is ok to make this a common rather than type-class definition. The IO instance is not doing anything special.

But call this `catches` not `catchesM`, and `Handler` not `HandlerM`.

And add a `{-# SPECIALISE #-}` pragma for `catchesHandler` and `catches` for `IO`.

Reply:
ok, I will.